AP_HAL_ChibiOS: add hwdef files for NarinFC-H7 (#29241)
ArduPilot/ardupilot: libraries/AP_HAL_ChibiOS/hwdef/NarinFC-H7/README.md

## Features/Specifications

-  **Processor**
    - STM32H743

-  **Sensors**
    - Accelerometer/Gyroscope: ADIS16470
    - Accelerometer/Gyroscope: ICM-20649
	- Accelerometer/Gyroscope: BMI088
	- Magnetometer: RM3100
	- Barometer: MS5611*2

-  **Interfaces**
    - 14 PWM servo outputs
	- Support multiple RC inputs (SBus / CPPM / DSM)
    - Analog/PWM RSSI input
	- 2 GPS ports (GPS and UART4 ports)
	- 4 ⅹ I2C buses
	- 2 ⅹ CAN bus ports
	- 2 ⅹ Power ports
	- 2 ⅹ ADC ports
	- 1 ⅹ USB port

-  **Power**
    - Power 4.3V ~ 5.4V
    - USB Input 4.75V ~ 5.25V

-  **Size and Dimensions**
    - 93.4mm x 46.4mm x 34.1mm
    - 106g

## Where to Buy

[VOLOLAND CO., LTD](https://vololand.com "VOLOLAND CO., LTD")

## Outline Dimensions

![Outline Dimensions](./images/2.Outline_Dimensions.png "Outline Dimensions")

## Wiring Diagram

![Wire Diagram](./images/3.Wire_Diagram.png "Wire Diagram")

## UART Mapping (Port Diagram & Pin outs)

 UART corresponding to each SERIAL port, and its default protocol, are shown below:

  - SERIAL0 = USB (MAVLink2 default)
  - SERIAL1 = USART2,Telemetry1 (MAVlink2 default,DMA-enabled) 
  - SERIAL2 = USART6,Telemetry2 (MAVlink2 default,DMA-enabled) 

>   - SERIAL3 = USART1,GPS1 (GPS default)

Reviewer: ```suggestion
  - SERIAL3 = USART1,GPS1 (GPS default, DMA-enabled)
```

  - SERIAL4 = UART4,GPS2 (GPS2 defualt)
  - SERIAL5 = UART8 (not available except on custom carrier boards)(USER default,DMA-enabled)

>   - SERIAL6 = UART7,DEBUG (USER,DMA-enabled)

Reviewer: ```suggestion
  - SERIAL6 = UART7,DEBUG (USER)
```

  - RSSI input

![SPI Port](./images/13.RSSI.png "RSSI input")

#### 6. GPS & Safety Port

![GPS & Safety Port](./images/4.5.GPS_Safety_Port_JST_GH_10P_Connector.png "GPS & Safety Port")

  - JST GH 10P connector
  - GPS NODMA

![Port Diagram & Pin outs](./images/4.Port_Diagram_Pin_outs_Diagram-B.png "Port Diagram-B")

#### 7. PWM & RC_IN

The NarinFC-H7 supports up to 14 PWM outputs. Outputs are grouped and all outputs within their group must be the same protocol.

![PWM Out](./images/4.6.PWM_Out_M1-M14.png "PWM Out")
  
  - 2.54mm pitch Dupont connector
  - RC_IN : Remote control receiver

#### 8. Power Input

![Power Input](./images/4.7.Power_Input.png "Power Input")

  - 2mm pitch Dupont connector

#### 9. ADC Port
  - ADC input

![SPI Port](./images/12.ADC.png "ADC input")

#### 10. DEBUG/UART7 Port
UART7(SERIAL6) is labeled DEBUG RX/TX below

![DEBUG Port](./images/4.8.DEBUG_Port_JST_HG_6P_Connector.png "DEBUG Port")
  
  - JST GH 6P connector
  - DEBUG NODMA

#### 11. USB Port
  - USB C Type

#### 12. SPI Port

![SPI Port](./images/4.10.SPI_Port_JST_GH_7P_Connector.png "SPI Port")

  - JST GH 7P connector
  - SPI Port

#### 13. SD CARD
  - SD CARD

## PWM Output

> The NarinFC-H7 supports up to 14 PWM outputs.All outputs except M13 and M14 support DShot. No outputs support Bi-Directional DShot.

Reviewer: ```suggestion
The NarinFC-H7 supports up to 14 PWM outputs.All outputs except M13 and M14 support DShot. Outputs 1-8 support Bi-Directional DShot.
```
